ocaml/xapi/storage_migrate_helper.mli

```diff
@@ -1,247 +0,0 @@
-
-    val t_leaf_dp : (string, t) Rpc.Types.field
-
-    val t_remote_dp : (string, t) Rpc.Types.field
-
-    val t_dest_sr : (Storage_interface.sr, t) Rpc.Types.field
-
-    val t_copy_vdi : (Storage_interface.vdi, t) Rpc.Types.field
-
-    val t_remote_url : (string, t) Rpc.Types.field
-
-    val t_verify_dest : (bool, t) Rpc.Types.field
-
-    val typ_of : t Rpc.Types.typ
-
-    val t : t Rpc.Types.def
-
-    val rpc_of_t : t -> Rpc.t
-
-    val t_of_rpc : Rpc.t -> t
-  end
-
-  val loaded : bool ref
-
-  val mutex : Mutex.t
-
-  type send_table = (string, Send_state.t) Hashtbl.t
-
-  type recv_table = (string, Receive_state.t) Hashtbl.t
-
-  type copy_table = (string, Copy_state.t) Hashtbl.t
-
-  type osend
-
-  type orecv
-
-  type ocopy
-
-  type _ operation =
-    | Send_op : Send_state.t -> osend operation
-    | Recv_op : Receive_state.t -> orecv operation
-    | Copy_op : Copy_state.t -> ocopy operation
-
-  type _ table =
-    | Send_table : send_table -> osend table
-    | Recv_table : recv_table -> orecv table
-    | Copy_table : copy_table -> ocopy table
-
-  val active_send : send_table
-
-  val active_recv : recv_table
-
-  val active_copy : copy_table
-
-  val table_of_op : 'a operation -> 'a table
-
-  val persist_root : string ref
-
-  val path_of_table : 'a table -> string
-
-  val rpc_of_table : 'a table -> Rpc.t
-
-  val to_string : 'a table -> string
-
-  val rpc_of_path : string -> Rpc.t
-
-  val save_one : 'a table -> unit
-
-  val save : unit -> unit
-
-  val access_table : save_after:bool -> ('a -> 'b) -> 'a -> 'b
-
-  val map_of :
-       unit
-    -> (string * Send_state.t) list
-       * (string * Receive_state.t) list
-       * (string * Copy_state.t) list
-
-  val add : string -> 'a operation -> unit
-
-  val find : 'a -> ('a, 'b) Hashtbl.t -> 'b option
-
-  val remove : 'a -> ('a, 'b) Hashtbl.t -> unit
-
-  val clear : unit -> unit
-
-  val remove_local_mirror : string -> unit
-
-  val remove_receive_mirror : string -> unit
-
-  val remove_copy : string -> unit
-
-  val find_active_local_mirror : string -> Send_state.t option
-
-  val find_active_receive_mirror : string -> Receive_state.t option
-
-  val find_active_copy : string -> Copy_state.t option
-
-  val mirror_id_of : Storage_interface.sr * Storage_interface.vdi -> string
-
-  val of_mirror_id : string -> Storage_interface.sr * Storage_interface.vdi
-
-  val copy_id_of : Storage_interface.sr * Storage_interface.vdi -> string
-
-  val of_copy_id : string -> Storage_interface.sr * Storage_interface.vdi
-end
-
-val vdi_info :
-  Storage_interface.async_result_t option -> Storage_interface.vdi_info
-
-val remove_from_sm_config :
-  Storage_interface.vdi_info -> string -> Storage_interface.vdi_info
-
-val add_to_sm_config :
-  Storage_interface.vdi_info -> string -> string -> Storage_interface.vdi_info
-
-val with_http :
-     Http.Request.t
-  -> (Http.Response.t * Unix.file_descr -> 'a)
-  -> Unix.file_descr
-  -> 'a
```

Comment:
Please add the .mli back with this at the end:

```ocaml
module Local : sig
  include module type of Storage_interface.StorageAPI (Idl.Exn.GenClient (struct
    let rpc call =
      Storage_utils.rpc ~srcstr:"smapiv2" ~dststr:"smapiv2"
        (Storage_utils.localhost_connection_args ())
        call
  end))
end

module type SMAPIv2 = module type of Local

val get_remote_backend : string -> bool -> (module SMAPIv2)
```
:)

And remove `module D : Debug.DEBUG`, which is unused